research/picai/nnunet_scripts/transfer_train.py

```diff
@@ -0,0 +1,277 @@
+        shutil.copy(
+            os.path.join(nnUNet_raw, ft_dataset_name, "dataset.json"),
+            os.path.join(nnUNet_preprocessed, ft_dataset_name, "dataset.json"),
+        )
+
+    return tl_plans_identifier
+
+
+def check_configs(id: int, plans_identifier: str, configs: list) -> None:
+    """Raises an error if configs are not found in plans json
+
+    Args:
+        id: The nnunet id of the dataset for which the plans were created
+        plans_identifier: The name of the plans file to check, not including
+            any extenstions
+        configs: A list of nnunet configs to check for
+    """
+    dataset_name = convert_id_to_dataset_name(id)
+    plans_path = os.path.join(nnUNet_preprocessed, dataset_name, plans_identifier + ".json")
+    plans = load_json(plans_path)
+    for c in configs:
+        assert (
+            c in plans["configurations"]
+        ), f"Did not find {c} config in {plans_identifier}.json. Note that \
+            3d_lowres and 3d_cascade_fullres are not created for datasets with \
+            small image sizes"
+
+
+def main() -> None:
+    """Parses arguments and opens subprocesses to train specified models
+
+    Notes: Subqequent trainings after the first model training must run in
+        subprocesses otherwise pytorch will refuse to use gpu for some reason.
+    """
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--ft_id", type=int, required=True, help="The nnUNet dataset ID integer for the fine-tuning dataset"
+    )
+    parser.add_argument(
+        "--pt_id",
+        type=int,
+        required=True,
+        help="[OPTIONAL] The dataset ID integer of the pretraining dataset"
+        ". Can be omitted if pt_fingerprint, pt_plans and pt_identifier"
+        "flags are all included",
+    )
+    parser.add_argument("--pt_plans_identifier", type=str, required=False, default="nnUNetPlans")
+    parser.add_argument(
+        "--pt_checkpoints",
+        type=str,
+        required=True,
+        nargs="+",
+        help="Path to the model checkpoints of the pretrained models."
+        " One checkpoint must be provided for each config and the order must be"
+        " consistent with the --configs flag",
+    )
+    parser.add_argument(
+        "--configs",
+        type=str,
+        nargs="+",
+        required=True,
+        help="Configuration that should be trained. Eg. '2d', '3d_fullres'."
+        "Multiple configs can be passed here and all will be trained "
+        "sequentially in the order provided given that a pretrained checkpoint"
+        " is given for each config. Note that 3d_cascade_fullres requires that"
+        " 3d_lowres already be trained first",
+    )
+    parser.add_argument(
+        "--fold", type=int, required=True, help="Fold of the 5-fold cross-validation. Should be an int between 0 and 4"
+    )
+    parser.add_argument(
+        "-np",
```

Comment:
nit: I know this may be carry over from nnUNet, but maybe expand this out to num_processes or something, np is definitely an overloaded acronym. there is also some variables below with np that we can expand out just to make everything very clear.